app/src/main/java/com/example/mystore/viewmodels/ProductViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.mystore.viewmodels
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.example.mystore.ProductsRepository
+import com.example.mystore.models.ProductModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class ProductViewModel @Inject constructor(
```

Comment:
Better to name this `ProductDetailsViewModel` so it's clearer that it's for that particular screen.